File: AdventTemplete.py
import os
import datetime

TAIL_FILE=("output.cpp","output.py","output.java","PuzzleInput.txt")
MAIN_ROOT="AdventOfCode"
YEAR=2024

def create_directory(directory):
    if not os.path.exists(directory):
        os.makedirs(directory)
        print("Directory created")
    else:
        print("Directory already exists")
def create_file(file_name):
    if not os.path.exists(file_name):
        with open(file_name, "w") as f:
            f.write("")
        print("File created")
    else:
        print("File already exists")
def createAllCalendars():
    for year in range(2015, datetime.datetime.now().year+1):
        dir_level1=MAIN_ROOT+"/Advent_of_code_"+str(year)
        create_directory(dir_level1)
        for date in range(1,25+1):
            dir_level2=dir_level1+"/Advent_of_code_"+str(year)+"_"+str(date)
            create_directory(dir_level2)
            #open this dir and add an empty file
            RootFile=dir_level2+"/Advent_of_code_"+str(year)+"_"+str(date)
            for tail in TAIL_FILE:
                create_file(RootFile+tail)
def createOneCalendar():
    dir_level1=MAIN_ROOT+"/Advent_of_code_"+str(YEAR)
    create_directory(dir_level1)
    for date in range(1,25+1):
        dir_level2=dir_level1+"/Advent_of_code_"+str(YEAR)+"_"+str(date)
        create_directory(dir_level2)
        #open this dir and add an empty file
        RootFile=dir_level2+"/Advent_of_code_"+str(YEAR)+"_"+str(date)
        for tail in TAIL_FILE:
            create_file(RootFile+tail)

File: AdventTemplete.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <sys/stat.h>
#include <sys/types.h>
#include <string>
#include <fstream>
#include <ctime>
//Put this file at VScode directory before runnine it
void create_dir(const std::string& path){
    const char* dir_name = path.c_str();
    if (mkdir(dir_name, 0777) == -1)
        std::cerr<<"Error :  "<< strerror(errno)<<"\n";
 
    else
        std::cout<<"Directory created\n";
}
void create_files(const std::string& dir_level2,const int& year, const int& date,std::string& tail){
    std::string dir_level3a=dir_level2+"/CPP_Advent_of_code"+std::to_string(year)+"_"+std::to_string(date)+tail;
    std::ofstream file(dir_level3a);
    file.close();
    std::cout<<"File created\n";
    std::cout << "File created : " << dir_level3a << std::endl;
}
int get_current_year(){
    time_t current_time;
    current_time = time(NULL);
    int currentYear = 1970 + current_time / 31537970; 
    return currentYear;
}
void createAllCalendars(const std::string& mainRoot, const std::vector<std::string>& tails){
    const int CurrenYear = get_current_year();
    for (int year=2015; year<=CurrenYear; year++){
        std::string dir_level1=mainRoot+"/Advent_of_code_"+std::to_string(year);
        create_dir(dir_level1);
        for(int date=1;date<=25;date++){
            std::string dir_level2 = dir_level1+"/Advent_of_code_"+std::to_string(year)+"_"+std::to_string(date);
            create_dir(dir_level2);
            for(std::string tail : tails){
                create_files(dir_level2,year,date,tail);
            }
        }
    }
}
void createOneCalendar(const std::string& mainRoot, const std::vector<std::string>& tails, const int& year){
    std::string dir_level1=mainRoot+"/Advent_of_code_"+std::to_string(year);
        create_dir(dir_level1);
        for(int date=1;date<=25;date++){
            std::string dir_level2 = dir_level1+"/Advent_of_code_"+std::to_string(year)+"_"+std::to_string(date);
            create_dir(dir_level2);
            for(std::string tail : tails){
                create_files(dir_level2,year,date,tail);
            }
        }
}
int main(){
    const std::vector<std::string> tails = {"output.cpp","output.py","output.java","PuzzleInput.txt"};
    const std::string mainRoot = "AdventOfCode";
    const int year  = 2024;
    createOneCalendar(mainRoot,tails,year);
}